pkg/integrations/v4/config/loader.go

```diff
@@ -0,0 +1,154 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"github.com/newrelic/infrastructure-agent/internal/integrations/v4/files"
+	"github.com/newrelic/infrastructure-agent/pkg/config/envvar"
+	"github.com/newrelic/infrastructure-agent/pkg/log"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+var clLog = log.WithComponent("integrations.config.Loader")
+
+// LegacyYAML is not an actual error. Used for discarding V3 plugins
+var LegacyYAML = errors.New("file format belongs to the old integrations format")
+
+const (
+	// Integrations V3 configs use the "instances" key word
+	// In the current Integrations config, we use "integrations"
+	// The two fields below will allows us distinguish between both, to ignore V4
+	integrationsField    = "integrations"
+	LegacyInstancesField = "instances"
+)
+
+// Loader will read and parse integrations v4 config files.
+type Loader interface {
+	// Load reads all the configuration files in a given directory. If path is a file instead of a directory
+	// will try to read it as a single configuration file.
+	Load(path string) (YAMLMap, error)
+
+	// LoadFile reads the configuration file and discards it if it's not belonging to the V4 format
+	LoadFile(file string) (ok bool, cfg YAML)
+}
+
+type pathLoader struct {
+	isDirectory func(path string) (bool, error)
+}
+
+// NewPathLoader returns a new instance of a config Loader.
+func NewPathLoader() Loader {
+	return &pathLoader{
+		isDirectory: isDirectory,
+	}
+}
+
+// Load reads all the configuration files in a given directory. If path is a file instead of a directory
+// will try to read it as a single configuration file.
+func (pl *pathLoader) Load(path string) (YAMLMap, error) {
+	isDir, err := isDirectory(path)
+	if err != nil {
+		return nil, err
+	}
+	if isDir {
+		return pl.loadDirectory(path)
+	}
+
+	ok, cfg := pl.LoadFile(path)
+	if !ok {
+		return nil, fmt.Errorf("no configuration found in path %s", path)
+	}
+
+	return YAMLMap{
+		path: cfg,
+	}, nil
+}
+
+// LoadFile reads the configuration file and discards it if it's not belonging to the V4 format
+func (pl *pathLoader) LoadFile(file string) (bool, YAML) {
+	fLog := clLog.WithField("file", file)
+
+	cfg, err := loadFile(file)
+	if err != nil {
```

Comment:
I think that this logic belongs to the manager. I would return an error and leave this error control in the manager